File: KOURPS/KOURPS/src/items/OzelKagit.java
package items;

public class OzelKagit extends Kagit {
    // Değişkenler
    public String isim;
    public int kalinlik;

    // Fonksiyonlar
    @Override
    public void nesneOzellikleriGoster() {
        super.nesneOzellikleriGoster();
        System.out.println("Nesne Ismi : " + isim);
        System.out.println("Kalınlık: " + kalinlik);
    }
    @Override
    public double etkiHesapla(double rakipSavunmasi) {
        return kalinlik * nufuz / rakipSavunmasi;
    }
    @Override
    public void durumGuncelle(double etki) {
        super.durumGuncelle(etki);
    }

    // Constructorlar
    public OzelKagit() {
        super();
        this.isim = "OzelKagit";
        this.kalinlik = 2;
        this.tasSavunma *= kalinlik;
        this.makasSavunma *= kalinlik;
    }
    public OzelKagit(int dayanma, int seviyeP, int nuf, int kalin) {
        super(dayanma, seviyeP, nuf);
        this.isim = "OzelKagit";
        this.kalinlik = kalin;
        this.tasSavunma *= kalinlik;
        this.makasSavunma *= kalinlik;
    }
}

File: Entities/objects/Concrete/OzelKagit.cs
using System;
namespace Entities.objects
{
    public class OzelKagit:Kagit
    {
        public int kalinlik;

        // Fonksiyonlar

        public void nesneOzellikleriGoster()
        {
            base.nesneOzellikleriGoster();
            Console.WriteLine("Kalınlık: " + kalinlik);
        }

        public double etkiHesapla(double rakipSavunmasi)
        {
            return kalinlik * nufuz / rakipSavunmasi;
        }

        public void durumGuncelle(double etki)
        {
            base.durumGuncelle(etki);
        }

        // Constructorlar
        public OzelKagit() : base()
        {
            this.kalinlik = 2;
            this.tasSavunma *= kalinlik;
            this.makasSavunma *= kalinlik;
        }
        public OzelKagit(int dayanma, int seviyeP, int nuf, int kalin) : base(dayanma, seviyeP, nuf)
        {
            this.kalinlik = kalin;
            this.tasSavunma *= kalinlik;
            this.makasSavunma *= kalinlik;
        }
    }
}

